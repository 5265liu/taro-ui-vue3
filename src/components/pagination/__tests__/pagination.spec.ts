import { mountFactory, Slots } from '@/tests/helper'
import { ref } from '@vue/runtime-core'
import AtPagination from '../index'

const factory = (values = {}, slots: Slots = { default: [] }) => {
  return mountFactory(AtPagination, {}, values, slots)
}

describe('AtPagination Snap', () => {
  it('render AtPagination -- props current', () => {
    const wrapper = factory({ current: 2 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtPagination -- props total', () => {
    const wrapper = factory({ total: 100 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtPagination -- props pageSize', () => {
    const wrapper = factory({ pageSize: 40 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtPagination -- props icon', () => {
    const wrapper = factory({ icon: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtPagination Event', () => {
  it('AtPagination onPageChange - prev', () => {
    const current = ref(2)
    const onPageChange = jest.fn().mockImplementation((val) => {
      current.value = val.current
    })
    const wrapper = factory({
      total: 100,
      current: current.value,
      onPageChange,
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.at-pagination .at-pagination__btn-prev .at-button').trigger('tap')
    expect(onPageChange).toBeCalled()
    expect(current.value).toBe(1)
  })

  it('AtPagination onPageChange - next', () => {
    const current = ref(2)
    const onPageChange = jest.fn().mockImplementation((val) => {
      current.value = val.current
    })
    const wrapper = factory({
      total: 100,
      current: current.value,
      onPageChange,
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.at-pagination .at-pagination__btn-next .at-button').trigger('tap')
    expect(onPageChange).toBeCalled()
    expect(current.value).toBe(3)
  })

  it('AtPagination onPageChange not to be called(disabled prev or next)', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      total: 20,
      onPageChange,
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.at-pagination .at-pagination__btn-prev .at-button').trigger('tap')
    wrapper.find('.at-pagination .at-pagination__btn-next .at-button').trigger('tap')
    expect(onPageChange).not.toBeCalled()
  })

  it('AtPagination onPageChange params {type, current} - prev', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      current: 2,
      total: 100,
      onPageChange,
    })
    wrapper.find('.at-pagination .at-pagination__btn-prev .at-button').trigger('tap')
    expect(onPageChange.mock.calls[0][0].type).toEqual('prev')
    expect(onPageChange.mock.calls[0][0].current).toEqual(1)
  })

  it('AtPagination onPageChange params {type, current} - next', () => {
    const onPageChange = jest.fn()
    const wrapper = factory({
      total: 100,
      onPageChange,
    })
    wrapper.find('.at-pagination .at-pagination__btn-next .at-button').trigger('tap')
    expect(onPageChange.mock.calls[0][0].type).toEqual('next')
    expect(onPageChange.mock.calls[0][0].current).toEqual(2)
  })
})
