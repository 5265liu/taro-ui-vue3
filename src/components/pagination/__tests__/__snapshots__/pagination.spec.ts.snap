// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtPagination Event AtPagination onPageChange - next 1`] = `
<view
  class="at-pagination"
>
  <view
    class="at-pagination__btn-prev"
  >
    <!---->
    <view
      class="at-button--small at-button"
    >
      <button
        class="at-button__wxbutton"
        lang="en"
      />
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        上一页
      </view>
    </view>
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      2
    </text>
    <text>
      /5
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <!---->
    <view
      class="at-button--small at-button"
    >
      <button
        class="at-button__wxbutton"
        lang="en"
      />
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        下一页
      </view>
    </view>
  </view>
</view>
`;

exports[`AtPagination Event AtPagination onPageChange - prev 1`] = `
<view
  class="at-pagination"
>
  <view
    class="at-pagination__btn-prev"
  >
    <!---->
    <view
      class="at-button--small at-button"
    >
      <button
        class="at-button__wxbutton"
        lang="en"
      />
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        上一页
      </view>
    </view>
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      2
    </text>
    <text>
      /5
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <!---->
    <view
      class="at-button--small at-button"
    >
      <button
        class="at-button__wxbutton"
        lang="en"
      />
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        下一页
      </view>
    </view>
  </view>
</view>
`;

exports[`AtPagination Event AtPagination onPageChange not to be called(disabled prev or next) 1`] = `
<view
  class="at-pagination"
>
  <view
    class="at-pagination__btn-prev"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        上一页
      </view>
    </view>
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      1
    </text>
    <text>
      /1
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        下一页
      </view>
    </view>
  </view>
</view>
`;

exports[`AtPagination Snap render AtPagination -- props current 1`] = `
<view
  class="at-pagination"
>
  <view
    class="at-pagination__btn-prev"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        上一页
      </view>
    </view>
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      2
    </text>
    <text>
      /1
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        下一页
      </view>
    </view>
  </view>
</view>
`;

exports[`AtPagination Snap render AtPagination -- props icon 1`] = `
<view
  class="at-pagination at-pagination--icon"
>
  <view
    class="at-pagination__btn-prev"
  >
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        <text
          class="at-icon at-icon-chevron-left"
        />
      </view>
    </view>
    <!---->
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      1
    </text>
    <text>
      /1
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        <text
          class="at-icon at-icon-chevron-right"
        />
      </view>
    </view>
    <!---->
  </view>
</view>
`;

exports[`AtPagination Snap render AtPagination -- props pageSize 1`] = `
<view
  class="at-pagination"
>
  <view
    class="at-pagination__btn-prev"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        上一页
      </view>
    </view>
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      1
    </text>
    <text>
      /1
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        下一页
      </view>
    </view>
  </view>
</view>
`;

exports[`AtPagination Snap render AtPagination -- props total 1`] = `
<view
  class="at-pagination"
>
  <view
    class="at-pagination__btn-prev"
  >
    <!---->
    <view
      class="at-button--small at-button--disabled at-button"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        上一页
      </view>
    </view>
  </view>
  <view
    class="at-pagination__number"
  >
    <text
      class="at-pagination__number-current"
    >
      1
    </text>
    <text>
      /5
    </text>
  </view>
  <view
    class="at-pagination__btn-next"
  >
    <!---->
    <view
      class="at-button--small at-button"
    >
      <button
        class="at-button__wxbutton"
        lang="en"
      />
      <!---->
      <!---->
      <!---->
      <view
        class="at-button__text"
      >
        下一页
      </view>
    </view>
  </view>
</view>
`;
